### PR TITLE
Adding trivial parallel and serial implementations

### PR DESCRIPTION
**Pull request by pandeyshubham25**
Generated scores look good to me. Tried on the sample graph in test directory.
Things to note -
1) We are not using threshold but number of iterations as stopping criteria
2) The two implementations (serial and trivial parallel) only output 3 things a) Parameter values b) Load time c) Iterations time to the console.
3) I decided to write both the page ids and their final scores. We can later parse whatever information we like during analysis as its always better to have extra information.

How to run -
g++ -fopenmp prank_serial.cpp -o prank_serial
./prank_serial 4 100 0.35 ../test/testdata

Lets not merge this branch at the moment, i strongly feel theres much to do in terms of code beautification :|

### CODE REVIEW

**Comment by NMerz** [comment]
Aside from the mostly duplicate files, I don't know that it is worth any beautification for a project this short. If you want a quick and dirty way to combine in one file, you can add a couple an extra define mode(s) to the make file (or I can if you want) like:
https://github.com/pandeyshubham25/pagerank/pull/6/files#diff-05890e6fc7ed8c96d26d42a367bed1b0cc6377f13fd450b9fb448383e47069a0R19-R23
and
https://github.com/pandeyshubham25/pagerank/pull/6/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R14-R30

**Comment by pandeyshubham25** [comment]
Agreed. Lets just focus on the main file then, as we are anyway using the other two just for generating results for comparison.

**Comment by NMerz** [comment]
I need more (slightly more) complicated build configs for Mac do you mind if cherry-pick over the Makefile commits from #6 ?
Or I can just keep them local if that would clutter this PR.

**Comment by NMerz** [comment]
I need more (slightly more) complicated build configs for Mac do you mind if cherry-pick over the Makefile commits from #6 ?
Or I can just keep them local if that would clutter this PR.

**Comment by pandeyshubham25** [comment]
Please go ahead and pick whatever helps you to run code, i dont mind!

File: src/prank_serial.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    cout<< "Iteration time: " << (double)(clock() - aStart)/CLOCKS_PER_SEC <<endl;
+
+    //PHASE 4: Write the result
+
+    string result_path = input_file + string("_serial_ranks") ;
+    ofstream result_file(result_path);
+    for(auto& x : pages){
+        result_file << x.first << " " << x.second.score << "\n";
+    }
+    result_file.close();
```
Comment (NMerz) [inline src/prank_serial.cpp]: Probably a brain fart on my part, but I cannot seem to find the resulting file from this when running:
`./prank_serial 9, 100, 0.850000, road_usa`

Comment (pandeyshubham25) [reply src/prank_serial.cpp]: is it printing rest of the output to console ? like time taken and stuff ?

Comment (NMerz) [reply src/prank_serial.cpp]: I get:
```
5.18571%
```
 The % is highlighted black (might mean some special?)

Comment (pandeyshubham25) [reply src/prank_serial.cpp]: This shouldnt be happening, lemme know if you can get on a call for this if you are free.. should be quick

Comment (NMerz) [reply src/prank_serial.cpp]: I can. One more quick data point: I must have messed something up when merging into eval. When I build on develop I get the following:
```
Loading graph data...
Graph data loaded in 5.2142s
Loaded 4682830 pages
```

Although I still don't see the file.

**Comment by NMerz** [comment]
I believe clock time is currently cpu time used.
Is this intentional or do we want wall time? Maybe both for efficiency calc?
https://www.cplusplus.com/reference/ctime/clock/

A couple times below:
```
(base) nmerz@Nathans-MacBook-Pro pagerank % time  ./prank_serial 9, 30, 0.850000, road_usa
Serial parameters: road_usa 30 0.85
Load time: 5.14273
Iteration time: 102.062
./prank_serial 9, 30, 0.850000, road_usa  109.09s user 0.33s system 99% cpu 1:49.54 total
```
```
(base) nmerz@Nathans-MacBook-Pro pagerank % time  ./prank_trivial_parallel 9, 30, 0.850000, road_usa 
Parallel parameters: road_usa 30 0.85
Load time: 5.18695
Iteration time: 126.189
./prank_trivial_parallel 9, 30, 0.850000, road_usa  130.91s user 2.94s system 580% cpu 23.041 total
```

**Comment by pandeyshubham25** [comment]
I see, lets do wall. lemme patch this with the floating point precision stuff.. got a little late. sorryy!

**Comment by pandeyshubham25** [comment]
Added my changes
1) Floating point precision set to 10
2) Clock time now using chrono library

Additionally, can you add gitignore to gitignore ? I have a different set of files in my local that i set to ignore and pulling your changes "disrupts" my gitignore :P

**Comment by NMerz** [comment]
> Additionally, can you add gitignore to gitignore ? I have a different set of files in my local that i set to ignore and pulling your changes "disrupts" my gitignore :P

Just add yours to the gitignore, more in there isn't harmful. I think it should be a tracked file though for executables, builds, etc.

**Comment by pandeyshubham25** [comment]
oh i see lol, i ll add mine next time too, i have some garbage coming from ide that i have put into it

**Pull request merged**